services/blockchain-indexer/shared/utils/snapshot.js

```diff
@@ -0,0 +1,222 @@
+							const unzip = zlib.createUnzip();
+							const writeFile = fs.createWriteStream(filePath);
+
+							res.pipe(unzip).pipe(writeFile);
+
+							res.on('error', (err) => {
+								reject(new Error(err));
+							});
+
+							writeFile.on('finish', () => {
+								// calculate hash from file.
+								calculateSHA256(filePath).then((calculatedChecksum) => {
+									if (calculatedChecksum === checksum) {
+										resolve();
+									} else {
+										reject(new Error('Checksum verification failed.'));
+									}
+								}).catch(err => {
+									reject(err);
+								});
+							});
+
+							writeFile.on('error', (err) => {
+								reject(err);
+							});
+						} else {
+							const errMessage = `Download failed with HTTP status code: ${res.statusCode} (${res.statusMessage}).`;
+							console.error(errMessage);
+							if (res.statusCode === 404) {
+								reject(new NotFoundException(errMessage));
+							} else {
+								reject(new Error(errMessage));
+							}
+						}
+					}).on('error', (err) => {
+						reject(new Error(err));
+					});
+				});
+			} else {
+				logger.error(`Failed to download the checksum file. HTTP status code: ${response.statusCode} (${response.statusMessage}).`);
+				reject(new Error('Failed to download the checksum file.'));
+			}
+		}).on('error', (err) => {
+			reject(new Error(err));
+		});
+	});
+};
+
+const resolveSnapshotRestoreCommand = async (connEndpoint) => {
+	await checkCommandAvailability();
+	const [user, password] = connEndpoint.split('//')[1].split('@')[0].split(':');
+	const [host, port, database] = connEndpoint.split('@')[1].split(new RegExp('/|:', 'g'));
+	const mysqlSnapshotCommand = `mysql ${database} -h ${host} -P ${port} -u ${user} -p${password} < ${snapshotFilePath}`;
+	return mysqlSnapshotCommand;
+};
+
+const applySnapshot = async (connEndpoint = MYSQL_ENDPOINT) => {
+	try {
+		logger.debug('Attempting to resolve the snapshot command.');
+		const snapshotRestoreCommand = await resolveSnapshotRestoreCommand(connEndpoint);
+		logger.info(`Attempting to apply the snapshot file available at: ${snapshotFilePath}.`);
+		const { stdout, stderr } = await execInShell(snapshotRestoreCommand);
+		logger.info(stdout);
+		logger.warn(stderr);
+		logger.info('SQL file(s) imported successfully.');
+	} catch (error) {
+		logger.error(error.stack);
+		throw error;
+	}
+};
+
+const downloadSnapshot = async (snapshotUrl) => {
+	const directoryPath = path.dirname(snapshotFilePath);
+	if (!(await exists(directoryPath))) await mkdir(directoryPath, { recursive: true });
+	await downloadUnzipAndVerifyChecksum(snapshotUrl, snapshotFilePath);
+};
+
+const initSnapshot = async () => {
+	if (config.snapshot.enable !== true) {
+		logger.info('Index snapshot application has been disabled.');
+		return;
+	}
+
+	const { chainID } = await requestConnector('getNetworkStatus');
+	const network = config.networks.LISK
+		.filter(networkInfo => networkInfo.chainID === chainID)[0];
```

Comment:
We can use find method here
```suggestion
	const network = config.networks.LISK
		.find(networkInfo => networkInfo.chainID === chainID);
```